rspace/src/main/scala/coop/rchain/rspace/HotStore.scala

```diff
@@ -286,25 +287,25 @@ private class InMemHotStore[F[_]: Sync, C, P, A, K](
 
 object HotStore {
 
-  def inMem[F[_]: Sync, C, P, A, K](
+  def inMem[F[_]: Concurrent, C, P, A, K](
       implicit S: Ref[F, Cache[C, P, A, K]],
-      HR: HistoryReader[F, C, P, A, K],
+      HR: RhoHistoryReader[F, C, P, A, K],
       ck: Codec[K]
   ): HotStore[F, C, P, A, K] =
     new InMemHotStore[F, C, P, A, K]
 
   def from[F[_], C, P, A, K](
       cache: Cache[C, P, A, K],
-      historyReader: HistoryReader[F, C, P, A, K]
-  )(implicit ck: Codec[K], sync: Sync[F]) =
+      historyReader: RhoHistoryReader[F, C, P, A, K]
+  )(implicit ck: Codec[K], concurrent: Concurrent[F]) =
```

Comment:
this Concurrent is not really necessary, but it will be required for HotStore refactoring (attempts are in https://github.com/rchain/rchain/pull/3271 and https://github.com/rchain/rchain/pull/3270), so I just left this in PR.

Reply:
I'm ok with this but in general what is not used it should be removed.